core/src/main/scala/org/apache/spark/scheduler/ShuffleMapTask.scala

```diff
@@ -69,7 +72,13 @@ private[spark] class ShuffleMapTask(
       val manager = SparkEnv.get.shuffleManager
       writer = manager.getWriter[Any, Any](dep.shuffleHandle, partitionId, context)
       writer.write(rdd.iterator(partition, context).asInstanceOf[Iterator[_ <: Product2[Any, Any]]])
-      return writer.stop(success = true).get
+      writer match {
+        case unsafe: UnsafeShuffleWriter[_, _] =>
```

Comment:
I guess we don't need to necessarily worry about the writer task failing due to an OutOfMemoryException because that will kill the executor.